core/src/telemetry/mod.rs

```diff
@@ -207,10 +218,53 @@ pub fn telemetry_init(opts: TelemetryOptions) -> Result<TelemetryInstance, anyho
                     Some(CoreLogConsumerLayer::new(consumer).with_filter(EnvFilter::new(filter)));
             }
         };
+
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .thread_name("telemetry")
+            .worker_threads(2)
+            .enable_all()
+            .build()
+            .unwrap();
+
+        // create otel export layer
+        runtime.block_on(async {
+            let tracer_cfg = opentelemetry_sdk::trace::Config::default()
+                .with_resource(default_resource_instance().clone());
+            let provider = opentelemetry_otlp::new_pipeline()
+                .tracing()
+                .with_exporter(
+                    opentelemetry_otlp::new_exporter()
+                        .tonic()
+                        .with_endpoint("grpc://localhost:4317".to_string()),
+                    // .with_metadata(MetadataMap::from_headers(headers.try_into()?)),
+                )
+                .with_trace_config(tracer_cfg)
+                // Using install_simple instead for now because install_batch is not producing spans and is emitting this error message:
+                // OpenTelemetry trace error occurred. cannot send message to batch processor as the channel is closed
+                // .install_batch(opentelemetry_sdk::runtime::Tokio)
+                .install_simple()
+                .unwrap();
+            opentelemetry::global::set_tracer_provider(provider.clone());
+
+            let tracer = provider.tracer_builder("sdk-core").build();
+            let opentelemetry = tracing_opentelemetry::layer().with_tracer(tracer);
+            //        .with_filter(EnvFilter::new(&tracing.filter))
+            export_layer = Some(opentelemetry);
+
+            let tracer = provider.tracer("sdk-core");
+
+            let _span = tracer
+                .span_builder("telemetry_init")
+                .with_kind(SpanKind::Server)
+                .with_attributes([KeyValue::new("temporal.worker", true)])
+                .start(&tracer);
+        });
```

Comment:
This span is being emitted, but the ones in `workflow_machines.rs` aren't.

Reply:
I am guessing this has to do with separate runtimes.

I can't remember exactly how I had this working with a separate one previously. Might need to review those diffs some more